go/database/mpt/verification.go

```diff
@@ -50,14 +93,21 @@ func VerifyFileForest(directory string, config MptConfig, roots []Root, observer
 	if observer == nil {
 		observer = NilVerificationObserver{}
 	}
+
 	observer.StartVerification()
-	defer func() {
-		if r := recover(); r != nil {
-			panic(r)
-		}
-		observer.EndVerification(res)
-	}()
+	defer observer.EndVerification(res)
```

Comment:
```suggestion
	defer observer.EndVerification(nil)
```

Reply:
I think the `res` here should remain as this is the reason why the interface accepts an error. If we want to remove it from here I suggest we remove the parameter all together.

Reply:
it looked to me that `res` is always `nil`. Is it correct? If yes, I do not see reason to pass a variable that is always nil, it only creates confusion.

Reply:
I dont think so. Any error return assigns the returned error to `res`

Reply:
but there is no error assigned to `res` - i.e. it is always nil.  For this reason I think it is rather confusing because it looks like an error is propagated, but it will never happen

Reply:
where is this test? I checked out this branch and could not find it. Anyway, when I copy and run it, it fails because of:
```

 verification.go:107: Unexpected call to *mpt.MockVerificationObserver.EndVerification([<nil>]) at /Users/kjezek/fantom/Carmen/go/database/mpt/verification.go:107 because: 
        expected call at /Users/kjezek/fantom/Carmen/go/database/mpt/verification_test.go:726 doesn't match the argument at index 0.
        Got: <nil> (<nil>)
        Want: not(is equal to <nil> (<nil>))

```

i.e. it proves the opposite

Reply:
So when the defer func is called this way
`defer observer.EndVerification(res)` - the test fails - its passed at the very beginning
but when its called this way
```
defer func(){
   defer observer.EndVerification(res)
}
```
 the test passes - its passed at the end when it gets assigned with `return err`

Reply:
[Here](https://stackoverflow.com/questions/47321012/difference-between-defer-call-a-function-directly-and-call-it-in-an-anonymous-fu) is an explanation.

Reply:
I got it, thanks for explanation.  It is confusing because it uses actually two not that know features:  a trick with correct defer function, and that the variable `res` gets assigned the return error.

I think it would help make this a bit less cryptic to name the return property `err` as well.

Reply:
I did not know about this defer feature as well. Anyhow I think `res` here means `result` as in a verification result is either error or nothing for success. We can rename it tho